Sources/TidepoolKit/TAPI.swift

```diff
@@ -95,71 +106,134 @@ public class TAPI {
-                        return DNSSRVRecord(priority: UInt16.max, weight: record.weight, host: record.host, port: record.port)
-                    }
-                    let environments = records.sorted().environments
-                    self.environmentsLocked.mutate { $0 = environments }
-                    self.logging?.debug("Successful DNS SRV record lookup")
-                    completion?(.success(environments))
-                }
-            }
+        // Lookup /info for current Tidepool environment, for issuer URL
+        let info = try await getInfo(environment: environment)
+
+        guard let issuer = info.auth?.issuerURL else {
+            throw TError.missingAuthenticationIssuer
         }
+
+        return try await getServiceConfiguration(issuer: issuer)
     }
 
     // MARK: - Authentication
 
-    /// Login to the Tidepool environment using the email and password. This is not typically invoked directly, but instead is
+    public func revokeTokens() async throws {
+        guard let session else {
+            throw TError.sessionMissing
+        }
+
+        // Lookup /info for current Tidepool environment, for issuer URL
+        let info = try await getInfo(environment: session.environment)
+
+        guard let issuer = info.auth?.issuerURL else {
+            throw TError.missingAuthenticationIssuer
+        }
+
+        let config = try await getServiceConfiguration(issuer: issuer)
+
+        guard let revokeURLStr = config.revocationEndpoint, let revokeURL = URL(string: revokeURLStr) else {
+            throw TError.missingAuthenticationConfiguration
+        }
+
+        try await revokeToken(revocationURL: revokeURL, token: session.accessToken, tokenType: "access_token")
+
+        if let refreshToken = session.refreshToken {
+            try await revokeToken(revocationURL: revokeURL, token: refreshToken, tokenType: "refresh_token")
+        }
+    }
+
+    private func revokeToken(revocationURL: URL, token: String, tokenType: String) async throws {
+
+        var bodyComponents = URLComponents()
+        bodyComponents.queryItems = [
+            URLQueryItem(name: "token", value: token),
+            URLQueryItem(name: "token_type_hint", value: tokenType),
+            URLQueryItem(name: "client_id", value: clientId),
+        ]
+
+        var request = URLRequest(url: revocationURL)
+        request.setValue("application/x-www-form-urlencoded", forHTTPHeaderField: "Content-Type")
+        request.httpMethod = "POST"
+        request.httpBody = bodyComponents.query?.data(using: .utf8)
+
+        let (data, response) = try await urlSession.data(for: request)
+
+        guard let response = response as? HTTPURLResponse else {
+            throw TError.responseUnexpected(response, data)
+        }
+
+        guard response.statusCode >= 200 && response.statusCode <= 299 else {
+            throw TError.responseUnexpectedStatusCode(response, data)
+        }
+
+        // Token has successfully been
```

Comment:
been what? Is this implementation incomplete or just the comment?

Reply:
Revoked. :) Will update. Thanks.